samples/security/protection_psoc6/unsecure/src/main.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, Cypress Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cy_sysint.h"
+#include <misc/printk.h>
+#include <zephyr.h>
+
+#include "cyprotection.h"
+
+#define CM0_MEMORY_ADDR (0x08000000 + 0x500)
+#define CM4_MEMORY_ADDR (CONFIG_SRAM_BASE_ADDRESS + 0x500)
```

Comment:
long term it might be good to have this as a memory region defined in DT.  This would make it less hard coded in the source and easily override-able via overlays.

Reply:
In this case, the defines are added exclusively to this example only. The two addresses needed: one from secure and another from non-secure memory regions, to show that the CM0 core can read from any memory, but CM4 - only from non-secure.

In general, it is needed to define at least one more sram region that will be used by the secure and non-secure world for sharing data (As it is done in ARM Mbed SPM). And both worlds should know this region addresses. But, at the moment it is not critical, and, actually, I do not know how to do this in Zephyr structure.
If needed - I can dig in the code and add several more special memory regions for normal SPM functioning.